pallets/parachain-staking/src/lib.rs

```diff
@@ -1844,14 +1879,19 @@ pub mod pallet {
 			<DelegatorState<T>>::get(acc).is_some()
 		}
 
-		/// Check whether an account is currently a collator candidate.
+		/// Check whether an account is currently a collator candidate and
+		/// whether their state is CollatorStatus::Active.
 		///
 		/// # <weight>
 		/// Weight: O(1)
 		/// - Reads: CollatorState
 		/// # </weight>
-		pub fn is_candidate(acc: &T::AccountId) -> bool {
-			<CollatorState<T>>::get(acc).is_some()
+		pub fn is_candidate_and_active(acc: &T::AccountId) -> (bool, bool) {
```

Comment:
could also be `is_active_candidate` and return an `Option<bool>`. I guess this would represent the exists or not and if it exists is active or not better.

Reply:
Much more elegant, thanks. See e547d10.